nemo/collections/asr/parts/feature_loader.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pickle
+
+import numpy as np
+import torch
+
+
+class ExternalFeatureLoader(object):
+    def __init__(self, file_path, sample_rate=16000, int_values=False, augmentor=None):
+        self._file_path = file_path
+        self.augmentor = augmentor
+
+    def load_feature_from_file(self, file_path):
+        # load pickle file
+        if file_path.endswith(".p") or file_path.endswith(".pickle"):
+            samples = pickle.load(open(file_path, "rb"))
```

Comment:
File not closed. Use with context for file and load into pickle

Reply:
good catch